buildpacks/gradle/src/errors.rs

```diff
@@ -0,0 +1,98 @@
+use crate::GradleBuildpackError;
+use buildpacks_jvm_shared::log::log_please_try_again_error;
+use indoc::{formatdoc, indoc};
+use libherokubuildpack::log::log_error;
+
+#[allow(clippy::too_many_lines, clippy::needless_pass_by_value)]
+pub(crate) fn on_error_gradle_buildpack(error: GradleBuildpackError) {
+    match error {
+        GradleBuildpackError::GradleWrapperNotFound => {
+            log_error(
+                "Missing Gradle Wrapper",
+                indoc! {"
+                    This buildpack leverages Gradle Wrapper to install the correct Gradle version to build your application.
+                    However, it seems that your application does not contain the required Gradle Wrapper files.
+
+                    To add the Gradle Wrapper, run the following command in your application's root directory:
+                    $ gradle wrapper
+
+                    Additional information about Gradle Wrapper and available configuration options can be found here:
+                    https://docs.gradle.org/current/userguide/gradle_wrapper.html
+                "},
+            );
+        }
+        GradleBuildpackError::GradleBuildIoError(error) => log_error(
+            "Failed to build app with Gradle",
+            formatdoc! {"
+                We're sorry this build is failing! If you can't find the issue in application code,
+                please submit a ticket so we can help: https://help.heroku.com/
+
+                Details: {error}
+            ", error = error },
+        ),
+        GradleBuildpackError::GradleBuildUnexpectedStatusError(exit_status) => {
+            let exit_code_string = exit_status.code().map_or_else(
+                || String::from("<unknown>"),
+                |exit_code| exit_code.to_string(),
+            );
+
+            log_error(
+                "Failed to build app with Gradle",
+                formatdoc! {"
+                    We're sorry this build is failing! If you can't find the issue in application code,
+                    please submit a ticket so we can help: https://help.heroku.com/
```

Comment:
(Future)
At some point it might be worth adjusting these "if in doubt open a ticket" messages to reduce the volume of tickets opened for things that aren't our fault / the tickets opened during an incident.

eg: Saying things like trying the build again, checking the status site, and then only after opening a ticket. 

Plus for failure modes where there is a 99.9% chance it's a customer issue, maybe not suggesting opening a ticket at all (they will still open a ticket if they really feel strongly).

Reply:
👍🏻 I agree.